modulus/datapipes/healpix/coupledtimeseries_dataset.py

```diff
@@ -294,6 +303,23 @@ def __getitem__(self, item):
                     else sol[self._input_indices[sample] + self._output_indices[sample]]
                 )
 
+        logger.log(5, "Adding gaussian noise to inputs and integrated_couplings")
+        if not self.forecast_mode and self.add_train_noise:
+            # Iterate over C: inputs.shape = [B, T, C, F, H, W]
+            for i in range(inputs.shape[2]):
+                inputs[:, :, i] += np.random.normal(
```

Comment:
I recommend adding an additional random seed parameter. This way we have fully reproducible training.